docusaurus/docs/how-to-guides/ui-extensions/versioning-extensions.md

```diff
@@ -0,0 +1,85 @@
+---
+id: versioning-extensions
+title: Version extensions
+description: This document covers best practices for versioning UI extensions, ensuring stability, compatibility, and smooth transitions.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - extensions
+  - ui-extensions
+sidebar_position: 40
+---
+
+# Best Practices for versioning UI Extensions
+
+To ensure stability and smooth transitions when updating UI extensions, use a _version suffix_ in the extension point (that is, the component ID). This approach helps maintain compatibility while allowing developers to introduce breaking changes in a controlled manner.
+
+## 1. **Use a Version Suffix in the ID**
+
+Each extension point ID/component ID should include a suffix indicating the major version of the extension.
+
+**Example:**
+
+```typescript
+// Initial version
+export const EXTENSION_POINT_OR_COMPONENT_ID_V1 = 'my-plugin-id/feature/v1';
+
+// Breaking change introduced
+export const EXTENSION_POINT_OR_COMPONENT_ID_V2 = 'my-plugin-id/feature/v2';
+```
+
+- Non-breaking changes (for example, adding optional properties) _do not_ require a new version suffix.
+- Breaking changes (for example, modifying behaviors or removing properties) _must_ introduce a new version suffix.
+
+## 2. **Support multiple versions during transition**
+
+When introducing a new major version, the application should serve both the old and new versions for a transition period. This allows consumers time to migrate without immediate disruptions.
+
+**Example:**
+
+- `my-plugin-id/feature/v1` continues to function while `my-plugin-id/feature/v2` is introduced.
+- Consumers gradually migrate to `v2`.
+- After a deprecation period, `v1` you can safely remove it.
+
+## 3. **Publish Types with Version Suffixes**
```

Comment:
@jackw @leventebalogh is it too early to mention the `@grafana/plugin-types` package?

Reply:
No I think it's the perfect time to start mentioning it. We need to add Levitate somewhere in the mix to prevent breaking changes in the NPM package but other than that I see no reason to hold off. We want plugin devs to start using the package.